pkg/controller/operators/olm/overrides/inject/inject_test.go

```diff
@@ -521,6 +541,120 @@ func TestInjectEnvIntoDeployment(t *testing.T) {
 	}
 }
 
+func TestInjectEnvFromIntoDeployment(t *testing.T) {
+	tests := []struct {
+		name       string
+		podSpec    *corev1.PodSpec
+		envFromVar []corev1.EnvFromSource
+		expected   *corev1.PodSpec
+	}{
+		{
+			// PodSpec has one container and `EnvFrom` is empty.
+			// Expected: All env variable(s) specified are injected.
+			name: "WithContainerHasNoEnvFromVar",
+			podSpec: &corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{},
+				},
+			},
+			envFromVar: defaultEnvFromVars,
+			expected: &corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						EnvFrom: defaultEnvFromVars,
+					},
+				},
+			},
+		},
+
+		{
+			// PodSpec has one container and it has non overlapping envFrom var(s).
+			// Expected: existing non overlapping env vars are intact.
+			name: "WithContainerHasNonOverlappingEnvFromVar",
+			podSpec: &corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						EnvFrom: []corev1.EnvFromSource{
+							corev1.EnvFromSource{
+								Prefix: "foo",
+							},
+						},
+					},
+				},
+			},
+			envFromVar: defaultEnvFromVars,
+			expected: &corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						EnvFrom: append([]corev1.EnvFromSource{
+							corev1.EnvFromSource{
+								Prefix: "foo",
+							},
+						}, defaultEnvFromVars...),
+					},
+				},
+			},
+		},
+		{
+			// PodSpec has more than one container(s)
+			// Expected: All container(s) should be updated as expected.
+			name: "WithMultipleContainers",
+			podSpec: &corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{},
+					corev1.Container{
+						EnvFrom: []corev1.EnvFromSource{
+							corev1.EnvFromSource{
+								Prefix: "foo",
+							},
+						},
+					},
+					corev1.Container{
+						EnvFrom: []corev1.EnvFromSource{
+							corev1.EnvFromSource{
+								Prefix: "bar",
+							},
+						},
+					},
+				},
+			},
+			envFromVar: defaultEnvFromVars,
+			expected: &corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						EnvFrom: defaultEnvFromVars,
+					},
+					corev1.Container{
+						EnvFrom: append([]corev1.EnvFromSource{
+							corev1.EnvFromSource{
+								Prefix: "foo",
+							},
+						}, defaultEnvFromVars...),
+					},
+					corev1.Container{
+						EnvFrom: append([]corev1.EnvFromSource{
+							corev1.EnvFromSource{
+								Prefix: "bar",
+							},
+						}, defaultEnvFromVars...),
+					},
+				},
+			},
+		},
+	}
+
```

Comment:
Can you add a test case for when the subscription config envFrom overlaps with the existing set of envFrom?

Reply:
Yes, I will add this test case and I found there are pointers in `envFrom` field. I will update the envFrom comparison logic as well.